src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -43,25 +65,45 @@ def __init__(
         self.hub_name = hub_name
         self.region = sagemaker_session.boto_region_name
         self._sagemaker_session = sagemaker_session
+        self._default_thread_pool_size = 20
+        self.hub_bucket_name = bucket_name or self._fetch_hub_bucket_name()
+        self._s3_client = self._get_s3_client()
+
+    def _get_s3_client(self) -> BaseClient:
+        """Returns an S3 client."""
+        return boto3.client("s3", region_name=self.region)
+
+    def _fetch_hub_bucket_name(self) -> str:
+        """Retrieves hub bucket name from Hub config if exists"""
+        try:
+            hub_response = self._sagemaker_session.describe_hub(hub_name=self.hub_name)
+            hub_bucket_prefix = hub_response["S3StorageConfig"].get("S3OutputPath", None)
+            if hub_bucket_prefix:
+                return hub_bucket_prefix.replace("s3://", "")
```

Comment:
I _think_ the `S3OutputPath` would be something like `s3://my-hub-bucket/my-hub` rather than `s3://my-hub-bucket/`. That would cause a problem because the "bucket name" would actually be a S3 prefix.

May also want to strip any trailing slashes